src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java

```diff
@@ -252,6 +252,158 @@ public void testDeleteUser() throws Exception {
+    assertEquals("Another error occurred in user3", error.getReason());
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    interceptor.getResponse().getRequest().getContent().writeTo(out);
+    JsonFactory jsonFactory = Utils.getDefaultJsonFactory();
+    GenericJson parsed = jsonFactory.fromString(new String(out.toByteArray()), GenericJson.class);
+    assertEquals(1, parsed.size());
+    List<Map<String, Object>> expected = ImmutableList.of(
+        user1.getProperties(jsonFactory),
+        user2.getProperties(jsonFactory),
+        user3.getProperties(jsonFactory)
+    );
+    assertEquals(expected, parsed.get("users"));
+  }
+
+  @Test
+  public void testImportUsersWithHash() throws Exception {
+    TestResponseInterceptor interceptor = initializeAppForUserManagement("{}");
+    ImportUserRecord user1 = ImportUserRecord.builder()
+        .setUid("user1")
+        .build();
+    ImportUserRecord user2 = ImportUserRecord.builder()
+        .setUid("user2")
+        .setPasswordHash("password".getBytes())
+        .build();
+
+    List<ImportUserRecord> users = ImmutableList.of(user1, user2);
+    UserImportHash hash = new UserImportHash("MOCK_HASH") {
+      @Override
+      protected Map<String, Object> getOptions() {
+        return ImmutableMap.<String, Object>of("key1", "value1", "key2", true);
+      }
+    };
+    UserImportResult result = FirebaseAuth.getInstance().importUsersAsync(users,
+        UserImportOptions.withHash(hash)).get();
+    checkRequestHeaders(interceptor);
+    assertEquals(2, result.getSuccessCount());
+    assertEquals(0, result.getFailureCount());
+    assertTrue(result.getErrors().isEmpty());
+
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    interceptor.getResponse().getRequest().getContent().writeTo(out);
+    JsonFactory jsonFactory = Utils.getDefaultJsonFactory();
+    GenericJson parsed = jsonFactory.fromString(new String(out.toByteArray()), GenericJson.class);
+    assertEquals(4, parsed.size());
+    List<Map<String, Object>> expected = ImmutableList.of(
+        user1.getProperties(jsonFactory),
+        user2.getProperties(jsonFactory)
+    );
+    assertEquals(expected, parsed.get("users"));
+    assertEquals("MOCK_HASH", parsed.get("hashAlgorithm"));
+    assertEquals("value1", parsed.get("key1"));
+    assertEquals(Boolean.TRUE, parsed.get("key2"));
+  }
+
+  @Test
+  public void testImportUsersMissingHash() {
+    initializeAppForUserManagement();
+    ImportUserRecord user1 = ImportUserRecord.builder()
+        .setUid("user1")
+        .build();
+    ImportUserRecord user2 = ImportUserRecord.builder()
+        .setUid("user2")
+        .setPasswordHash("password".getBytes())
+        .build();
+
+    List<ImportUserRecord> users = ImmutableList.of(user1, user2);
+    try {
+      FirebaseAuth.getInstance().importUsersAsync(users);
+      fail("No error thrown for missing hash option");
+    } catch (IllegalArgumentException expected) {
+      // expected
```

Comment:
Please verify the error message. That will also make it super obvious what this test is verifying.

Reply:
Done